Latexify with various functions of states/params broken

```julia
using ModelingToolkit, Latexify

ex = :((1+cos(t))/(1+2*x))
latexify(ex) #works

@parameters t
@variables x(t)
@derivatives D'~t
eqs = [D(x) ~ (1+cos(t))/(1+2*x)]
latexify(eqs)
```
ModelingToolkit v4.3.0 with Latexify v0.14.6 produces
```
ERROR: LoadError: AssertionError: latexify does not support objects of type typeof(cos).
Stacktrace:
 [1] latexraw(::Function; kwargs::Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:convert_unicode, :env),Tuple{Bool,Symbol}}}) at /home/abc/.julia/packages/Latexify/gaIt2/src/latexraw.jl:89
 [2] (::Latexify.var"#22#25"{Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:convert_unicode, :env),Tuple{Bool,Symbol}}}})(::Function) at /home/abc/.julia/packages/Latexify/gaIt2/src/latexoperation.jl:11
 [3] iterate at ./generator.jl:47 [inlined]
 [4] _collect(::Array{Any,1}, ::Base.Generator{Array{Any,1},Latexify.var"#22#25"{Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:convert_unicode, :env),Tuple{Bool,Symbol}}}}}, ::Base.EltypeUnknown, ::Base.HasShape{1}) at ./array.jl:699
 [5] collect_similar at ./array.jl:628 [inlined]
 [6] map at ./abstractarray.jl:2162 [inlined]
 [7] latexoperation(::Expr, ::Array{Symbol,1}; cdot::Bool, index::Symbol, kwargs::Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:convert_unicode, :env),Tuple{Bool,Symbol}}}) at /home/abc/.julia/packages/Latexify/gaIt2/src/latexoperation.jl:11
 [8] (::Latexify.var"#recurseexp!#6"{Bool,Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:cdot, :env),Tuple{Bool,Symbol}}}})(::Expr) at /home/abc/.julia/packages/Latexify/gaIt2/src/latexraw.jl:79
 [9] (::Latexify.var"#recurseexp!#6"{Bool,Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:cdot, :env),Tuple{Bool,Symbol}}}})(::Expr) at /home/abc/.julia/packages/Latexify/gaIt2/src/latexraw.jl:74 (repeats 2 times)
 [10] latexraw(::Expr; convert_unicode::Bool, kwargs::Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:cdot, :env),Tuple{Bool,Symbol}}}) at /home/abc/.julia/packages/Latexify/gaIt2/src/latexraw.jl:82
 [11] #9 at ./none:0 [inlined]
 [12] iterate at ./generator.jl:47 [inlined]
 [13] collect_to! at ./array.jl:732 [inlined]
 [14] collect_to_with_first!(::Array{LaTeXStrings.LaTeXString,2}, ::LaTeXStrings.LaTeXString, ::Base.Generator{Array{Expr,2},Latexify.var"#9#10"{Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:cdot, :env),Tuple{Bool,Symbol}}}}}, ::Int64) at ./array.jl:710
 [15] collect(::Base.Generator{Array{Expr,2},Latexify.var"#9#10"{Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:cdot, :env),Tuple{Bool,Symbol}}}}}) at ./array.jl:691
 [16] latexraw(::Array{Expr,2}; kwargs::Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:cdot, :env),Tuple{Bool,Symbol}}}) at /home/abc/.julia/packages/Latexify/gaIt2/src/latexraw.jl:92
 [17] latexalign(::Array{Expr,2}; separator::String, double_linebreak::Bool, starred::Bool, rows::Symbol, kwargs::Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:cdot, :env),Tuple{Bool,Symbol}}}) at /home/abc/.julia/packages/Latexify/gaIt2/src/latexalign.jl:46
 [18] latexalign(::Tuple{Array{Expr,1},Array{Expr,1}}; kwargs::Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:cdot, :env),Tuple{Bool,Symbol}}}) at /home/abc/.julia/packages/Latexify/gaIt2/src/latexalign.jl:79
 [19] latexify(::Array{Equation,1}; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /home/abc/.julia/packages/Latexify/gaIt2/src/latexify_function.jl:11
 [20] latexify(::Array{Equation,1}) at /home/abc/.julia/packages/Latexify/gaIt2/src/latexify_function.jl:4
 [21] top-level scope at /home/abc/latexify_try.jl:18
 [22] include_string(::Function, ::Module, ::String, ::String) at ./loading.jl:1091
 [23] invokelatest(::Any, ::Any, ::Vararg{Any,N} where N; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at ./essentials.jl:710
 [24] invokelatest(::Any, ::Any, ::Vararg{Any,N} where N) at ./essentials.jl:709
 [25] inlineeval(::Module, ::String, ::Int64, ::Int64, ::String; softscope::Bool) at /home/abc/.vscode-server/extensions/julialang.language-julia-1.0.10/scripts/packages/VSCodeServer/src/eval.jl:185
 [26] (::VSCodeServer.var"#61#65"{String,Int64,Int64,String,Module,Bool,VSCodeServer.ReplRunCodeRequestParams})() at /home/abc/.vscode-server/extensions/julialang.language-julia-1.0.10/scripts/packages/VSCodeServer/src/eval.jl:144
 [27] withpath(::VSCodeServer.var"#61#65"{String,Int64,Int64,String,Module,Bool,VSCodeServer.ReplRunCodeRequestParams}, ::String) at /home/abc/.vscode-server/extensions/julialang.language-julia-1.0.10/scripts/packages/VSCodeServer/src/repl.jl:124
 [28] (::VSCodeServer.var"#60#64"{String,Int64,Int64,String,Module,Bool,Bool,VSCodeServer.ReplRunCodeRequestParams})() at /home/abc/.vscode-server/extensions/julialang.language-julia-1.0.10/scripts/packages/VSCodeServer/src/eval.jl:142
 [29] hideprompt(::VSCodeServer.var"#60#64"{String,Int64,Int64,String,Module,Bool,Bool,VSCodeServer.ReplRunCodeRequestParams}) at /home/abc/.vscode-server/extensions/julialang.language-julia-1.0.10/scripts/packages/VSCodeServer/src/repl.jl:36
 [30] (::VSCodeServer.var"#59#63"{String,Int64,Int64,String,Module,Bool,Bool,VSCodeServer.ReplRunCodeRequestParams})() at /home/abc/.vscode-server/extensions/julialang.language-julia-1.0.10/scripts/packages/VSCodeServer/src/eval.jl:110
 [31] with_logstate(::Function, ::Any) at ./logging.jl:408
 [32] with_logger at ./logging.jl:514 [inlined]
 [33] (::VSCodeServer.var"#58#62"{VSCodeServer.ReplRunCodeRequestParams})() at /home/abc/.vscode-server/extensions/julialang.language-julia-1.0.10/scripts/packages/VSCodeServer/src/eval.jl:109
 [34] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [35] invokelatest(::Any) at ./essentials.jl:709
 [36] macro expansion at /home/abc/.vscode-server/extensions/julialang.language-julia-1.0.10/scripts/packages/VSCodeServer/src/eval.jl:27 [inlined]
 [37] (::VSCodeServer.var"#56#57")() at ./task.jl:356
```

I have an old environment and this does work on ModelingToolkit v3.17.0

--- a/src/latexify_recipes.jl
+++ b/src/latexify_recipes.jl
@@ -1,5 +1,5 @@
 prettify_expr(expr) = expr
-prettify_expr(f::Function) = Base.isoperator(nameof(f)) ? nameof(f) : f
+prettify_expr(f::Function) = nameof(f)
 prettify_expr(expr::Expr) = Expr(expr.head, prettify_expr.(expr.args)...)
 
 @latexrecipe function f(eqs::Vector{ModelingToolkit.Equation})

--- a/test/latexify.jl
+++ b/test/latexify.jl
@@ -26,27 +26,29 @@ eqs = [D(x) ~ σ*(y-x)*D(x-y)/D(z),
        0 ~ σ*x*(ρ-z)/10-y,
        D(z) ~ x*y^(2//3) - β*z]
 
-@test latexify(eqs) == raw"""\begin{align}
+
+# Latexify.@generate_test latexify(eqs)
+@test latexify(eqs) == replace(
+raw"\begin{align}
 \frac{dx(t)}{dt} =& \frac{\sigma \left( y\left( t \right) - x\left( t \right) \right) \frac{d\left(x\left( t \right) - y\left( t \right)\right)}{dt}}{\frac{dz(t)}{dt}} \\
 0 =& \frac{\sigma x\left( t \right) \left( \rho - z\left( t \right) \right)}{10} - y\left( t \right) \\
 \frac{dz(t)}{dt} =& x\left( t \right) \left( y\left( t \right) \right)^{\frac{2}{3}} - \beta z\left( t \right)
 \end{align}
-"""
-@parameters t p[1:3]
-@variables u[1:3](t)
-@derivatives D'~t
+", "\r\n"=>"\n")
 
+@variables u[1:3](t)
+@parameters p[1:3]
 eqs = [D(u[1]) ~ p[3]*(u[2]-u[1]),
        0 ~ p[2]*p[3]*u[1]*(p[1]-u[1])/10-u[2],
        D(u[3]) ~ u[1]*u[2]^(2//3) - p[3]*u[3]]
 
-@test latexify(eqs) ==
+@test latexify(eqs) == replace(
 raw"\begin{align}
 \frac{du{_1}(t)}{dt} =& p{_3} \left( \mathrm{u{_2}}\left( t \right) - \mathrm{u{_1}}\left( t \right) \right) \\
 0 =& \frac{p{_2} p{_3} \mathrm{u{_1}}\left( t \right) \left( p{_1} - \mathrm{u{_1}}\left( t \right) \right)}{10} - \mathrm{u{_2}}\left( t \right) \\
 \frac{du{_3}(t)}{dt} =& \mathrm{u{_1}}\left( t \right) \left( \mathrm{u{_2}}\left( t \right) \right)^{\frac{2}{3}} - p{_3} \mathrm{u{_3}}\left( t \right)
 \end{align}
-"
+", "\r\n"=>"\n")
 
 eqs = [D(u[1]) ~ p[3]*(u[2]-u[1]),
        D(u[2]) ~ p[2]*p[3]*u[1]*(p[1]-u[1])/10-u[2],
@@ -54,10 +56,22 @@ eqs = [D(u[1]) ~ p[3]*(u[2]-u[1]),
 
 sys = ODESystem(eqs)
 
-@test latexify(eqs) ==
+@test latexify(eqs) == replace(
 raw"\begin{align}
 \frac{du{_1}(t)}{dt} =& p{_3} \left( \mathrm{u{_2}}\left( t \right) - \mathrm{u{_1}}\left( t \right) \right) \\
 \frac{du{_2}(t)}{dt} =& \frac{p{_2} p{_3} \mathrm{u{_1}}\left( t \right) \left( p{_1} - \mathrm{u{_1}}\left( t \right) \right)}{10} - \mathrm{u{_2}}\left( t \right) \\
 \frac{du{_3}(t)}{dt} =& \mathrm{u{_1}}\left( t \right) \left( \mathrm{u{_2}}\left( t \right) \right)^{\frac{2}{3}} - p{_3} \mathrm{u{_3}}\left( t \right)
 \end{align}
-"
+", "\r\n"=>"\n")
+
+
+@parameters t
+@variables x(t)
+@derivatives D'~t
+eqs = [D(x) ~ (1+cos(t))/(1+2*x)]
+
+@test latexify(eqs) == replace(
+raw"\begin{align}
+\frac{dx(t)}{dt} =& \frac{1 + \cos\left( t \right)}{1 + 2 x\left( t \right)}
+\end{align}
+", "\r\n"=>"\n")

